### ISSUE TITLE
Returning 404 from a component sets incorrect headers

### ISSUE THREAD

**Issue by sozonovalexey** [octobercms/october#2634]
##### Expected behavior
Headers:
Status Code:404 Not Found

##### Actual behavior
Headers:
Status Code:200 OK

##### Reproduce steps
1. Create new component

~~~
class Something extends ComponentBase
{
    public function onRun()
    {
        $this->setStatusCode(404);
        return $this->controller->run('404');
    }
}
~~~

2. Open page with new component and check headers

##### October build
393

**Comment by sozonovalexey** [comment]
It seems that everything is fine for build: 361

**Comment by arzola** [comment]
I think you have to put this on the init method

**Comment by daftspunk** [comment]
This should have been fixed in 2350e30853cc2c1337697bb38c113f7319eea0f1 (Build 319+)
